### ISSUE TITLE
feat(eds/store): Shard recovery or eviction on recognized data corruption

### ISSUE THREAD

**Issue by distractedm1nd** [celestiaorg/celestia-node#2609]
If Shards get corrupted or deleted, they will still persist in the dagstore until they are explicitly removed. This means any call to GetEDS or GetSharesByNamespace to that block will fail, because it will not get past StoreGetter. By removing the shard upon corruption detection, the next call to GetEDS will recover and store the block. Long term, we need to implement CAR recovery from the non-corrupted erasure coded data in the file. 

This PR will also require making the Bridge's share.Getter a Tee(Cascade(Store, Shrex)), so that bridges have a possibility to redownload and store the block post-removal.

This issue is made up of two tasks:

```[tasklist]
### Tasks
- [x] Destruction of shards on `OpShardFail` from dagstore's failure channel, and destruction upon nmt node not found after CollectSharesByNamespace call in StoreGetter's GetSharesByNamespace
- [ ] Implementation of CAR recovery via EDS reconstruction through erasure coding
```

**Comment by distractedm1nd** [comment]
Related: This would fix issues similar to #2335

**Comment by distractedm1nd** [comment]
@walldiss I am closing this issue since we are removing CAR, but I'm tagging you in case there is a schwap equivalent you want to add to schwap tracking issue
